Sources/Amplitude/Utilities/HttpClient.swift

```diff
@@ -9,13 +9,17 @@ import Foundation
 
 class HttpClient {
     let configuration: Configuration
-    internal var session: URLSession
+    internal let session: URLSession
 
     init(configuration: Configuration) {
         self.configuration = configuration
-        // shared instance has limitations but think we are not affected
-        // https://developer.apple.com/documentation/foundation/urlsession/1409000-shared
-        self.session = URLSession.shared
+        // an exclusive ephemeral session avoids contention (fewer request timeouts)
+        let sessionConfiguration = URLSessionConfiguration.ephemeral
```

Comment:
How does `ephermal` prevent timeouts? I am concerned that since `ephermal` doesn't use disk, a large event payload could consume a lot of RAM.

Reply:
It's not so much 'ephemeral' part, but the 'exclusive session' part that avoids contention. i.e. instantiating our own URLSession rather than using the `shared` class instance.

`URLSession.shared` could be used elsewhere by the app and be contended by other requests. By creating our own exclusive session we have a bit more control of ensuring we have an outside line, plus it gives us the opportunity to customise the session more precisely to our needs.

Having said that, default` is probably a better choice than `ephemeral`, but we should also eliminate caching entirely by also setting [`urlCache`](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/1410148-urlcache) to `nil`.